apps/src/applab/firebaseCounters.js

```diff
@@ -69,6 +69,45 @@ export function updateTableCounters(tableName, rowCountChange, updateNextId) {
 }
 
 /**
+ * Increment or reset the rate-limiting counters associated with the interval.
+ * @param {number} maxWriteCount
+ * @param {string} interval
+ * @param {number} currentTimeMs
+ * @returns {Promise}
+ */
+function incrementIntervalCounters(maxWriteCount, interval, currentTimeMs) {
+  const limitRef = getDatabase(Applab.channelId).child(`counters/limits/${interval}`);
+  return limitRef.transaction(limitData => {
+    limitData = limitData || {};
+    limitData.lastResetTime = limitData.lastResetTime || 0;
+    limitData.writeCount = (limitData.writeCount || 0) + 1;
+    if (limitData.writeCount <= maxWriteCount) {
+      return limitData;
+    } else if (limitData.lastResetTime + interval * 1000 < currentTimeMs) {
```

Comment:
Oops -- I got lucky that javascript let's you do things like `'15' * 1000` --> `15000`. I've extracted this as `const intervalMs = Number(interval) * 1000`.